### PR TITLE
Revert functorial state of runNonDet to []

### PR DESCRIPTION
**Pull request by KingoftheHomeless**
See #246 

all around me are familiar faces

### CODE REVIEW

**Comment by isovector** [comment]
i find this hard to take :(

File: test/AlternativeSpec.hs [hunk 1]
```diff
@@ -38,3 +45,11 @@ spec = parallel $ do
         `shouldBe` Just (["salabim"], ())
       (run . runTraceList . runNonDetMaybe) failtrace'
         `shouldBe` (["sim", "salabim"], Just ())
+
+  describe "runNonDet" $ do
+    it "should have terrible semantics when <|> is used within\
```
Comment (isovector) [inline test/AlternativeSpec.hs]: excellent test name

Comment (isovector) [reply test/AlternativeSpec.hs]: could we also write an "expect failure" for what it should be, just to document that?

Comment (KingoftheHomeless) [reply test/AlternativeSpec.hs]: Good idea. Simply via `shouldNotBe`, or is there something better?

Comment (isovector) [reply test/AlternativeSpec.hs]: There's nothing better :( but god would it be nice if there were.

**Pull request merged**